### PR TITLE
fix navigator is undefined in server side rendering mode

### LINKED ISSUE #569: angular directive fails in server-side-rendering (SSR) mode
**Describe the bug**
Angular directive fails in server-side rendering mode whit the following error : 
```
   Error: Uncaught (in promise): ReferenceError: navigator is not defined
   ReferenceError: navigator is not defined
       at IMaskDirective._isAndroid (/var/www/dist/app/fr-FR/webpack:/node_modules/angular-imask/fesm2015/angular-imask.js:279:1)
       at new IMaskDirective (/var/www/dist/app/fr-FR/webpack:/node_modules/angular-imask/fesm2015/angular-imask.js:86:1)
       at NodeInjectorFactory.IMaskDirective_Factory [as factory] (/var/www/dist/app/fr-FR/webpack:/node_modules/angular-imask/fesm2015/angular-imask.js:282:60)
```

**To Reproduce**
- create a new angular project
```
ng new imask-ssr
cd imask-ssr
```
add angular/universal
```
ng add @nguniversal/express-engine
```
add imask, import IMaskModule in app.module.ts and use [imask] directive in app.component
```
npm install --save angular-imask
```

```
npm run build:ssr
npm run serve:ssr
```
reach http://localhost:4000 to have the error

reproduced in repository : https://github.com/b-legrand/imask-ssr 

**Expected behavior**
No error when using angular universal

**Environment:**
 - IMask version
 - Framework/plugin version if used (angular 12

**Additional context**
`navigator` object is not defined in SSR, it is like `window` and `document` and should be considered optional

a simple fix should be to return false on _isAndroid when platform is not browser
a more advanced fix would be to allow users of the lib to provide user-agent from external source ( like [ngx-device-detector](https://www.npmjs.com/package/ngx-device-detector) for instance)

### PR DESCRIPTION
**Pull request by b-legrand**
- secure access to navigator in server mode
- fixes #569

### CODE REVIEW

File: packages/angular-imask/src/imask.directive.ts [hunk 1]
```diff
@@ -190,6 +192,10 @@ export class IMaskDirective<Opts extends IMask.AnyMaskedOptions> implements Cont
   }
 
   private _isAndroid(): boolean {
+    // secure access to navigator on server renders
+    if (isPlatformServer(this._platformId)) {
+      return false;
+    }
     return /android (\d+)/.test(navigator.userAgent.toLowerCase());
```
Comment (waterplea) [inline packages/angular-imask/src/imask.directive.ts:199]: You can one-line this as following:
```ts
return isPlatrformBrowser(this._platformId) && /android (\d+)/.test(navigator.userAgent.toLowerCase());
```

Comment (b-legrand) [reply packages/angular-imask/src/imask.directive.ts]: yes, done.

https://github.com/uNmAnNeR/imaskjs/pull/570/commits/35b9069832270bcdca0671436754848b53e55202#diff-5a0dc28acf4f1e69d882fe8fc38c3c55ac1d21a2a5b3a18144f498ac8926f6b7R195

**Comment by waterplea** [comment]
@uNmAnNeR will you give this a look? This issue severely limits use of iMask in serious Angular apps.

File: packages/angular-imask/src/imask.directive.ts (hunk 1 above)
Comment (waterplea) [reply packages/angular-imask/src/imask.directive.ts]: It's either `!isPlatformServer` or `isPlafromBrowser` :)

Comment (b-legrand) [reply packages/angular-imask/src/imask.directive.ts]: thanks, read too fast

**Pull request merged**